SMLHelper/Patchers/LootDistributionPatcher.cs

```diff
@@ -0,0 +1,119 @@
+﻿namespace SMLHelper.V2.Patchers
+{
+    using System; 
+    using Harmony;
+    using System.Collections.Generic;
+    using UnityEngine;
+    using UWE;
+    using Logger = V2.Logger;
+
+    internal class LootDistributionPatcher
+    {
+        internal static SelfCheckingDictionary<string, LootDistributionData.SrcData> CustomSrcData = new SelfCheckingDictionary<string, LootDistributionData.SrcData>("CustomSrcData");
+
+        internal static void Patch(HarmonyInstance harmony)
+        {
+            harmony.Patch(AccessTools.Method(typeof(LootDistributionData), "Initialize"),
+                postfix: new HarmonyMethod(AccessTools.Method(typeof(LootDistributionPatcher), "InitializePostfix")));
+
+            Logger.Log("LootDistributionPatcher is done.", LogLevel.Debug);
+        }
+
+        private static void InitializePostfix(LootDistributionData __instance)
+        {
+            foreach(var entry in CustomSrcData)
+            {
+                if(__instance.srcDistribution.ContainsKey(entry.Key))
```

Comment:
`srcDistribution` is a Dictionary.  
Use [TryGetValue](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=netframework-4.8) instead of a ContainsKey followed by another key lookup when accessing by index.

```CS
if (__instance.srcDistribution.TryGetValue(entry.Key, out LootDistributionData.SrcData srcData))
```